gui/src/mainwindow.cpp

```diff
@@ -0,0 +1,1459 @@
+                                                    lastAccessedDirectory,
+                                                    tr("Config file (.cfg)"));
+    exportConfig(filepath);
+}
+
+void MainWindow::on_localImageFileBrowseButton_clicked()
+{
+    QString path = QFileDialog::getOpenFileName(this,
+                                                tr("Select Image File"),
+                                                lastAccessedDirectory,
+                                                tr("Image files (*.tif *.jpg *.JPEG *.png *.bmp);;All files (*.*)"));
+    lastAccessedDirectory = "";
+
+    if(!path.isEmpty() && !path.isNull()) {
+        ui->localImageFileLineEdit->setText(path);
+        //manually invoke the slot to check the new filepath
+        on_imagepathLineEditLostFocus();
+    }
+}
+
+void MainWindow::on_modelFileBrowseButton_clicked()
+{
+    QString path = QFileDialog::getOpenFileName(this,
+                                                tr("Select Model File"),
+                                                lastAccessedDirectory,
+                                                tr("GBDXM files (*.gbdxm);;All files (*.*)"));
+    //Setting the last model directory to an empty string ensures that the browser will open
+    //in the last directory it accessed the last time it was opened
+    lastAccessedDirectory = "";
+
+    //The directory path string will be empty if the user presses cancel in the QFileDialog
+    if(!path.isEmpty() && !path.isNull()) {
+        ui->modelFileLineEdit->setText(path);
+        //manually invoke the slot to check the new filepath
+        on_modelpathLineEditLostFocus();
+    }
+}
+
+void MainWindow::on_imageSourceComboBox_currentIndexChanged(const QString &source)
+{
+    if(source != "Local Image File") {
+        //bbox
+        ui->bboxOverrideCheckBox->hide();
+        ui->bboxNorthLineEdit->setEnabled(true);
+        ui->bboxSouthLineEdit->setEnabled(true);
+        ui->bboxEastLineEdit->setEnabled(true);
+        ui->bboxWestLineEdit->setEnabled(true);
+
+        //local image selection
+        ui->localImageFileLabel->setEnabled(false);
+        ui->localImageFileLineEdit->setEnabled(false);
+        ui->localImageFileBrowseButton->setEnabled(false);
+
+        //set the style of the local image file field to default
+        ui->localImageFileLineEdit->clear();
+        ui->localImageFileLineEdit->setStyleSheet("color: default");
+
+        //token
+        ui->tokenLabel->setEnabled(true);
+        ui->tokenLineEdit->setEnabled(true);
+
+        //map id
+        if(source == "MapsAPI") {
+        	ui->mapIdLabel->setEnabled(true);
+        	ui->mapIdLineEdit->setEnabled(true);
+
+        	//user credentials
+        	ui->usernameLabel->setEnabled(false);
+        	ui->usernameLineEdit->setEnabled(false);
+        	ui->passwordLabel->setEnabled(false);
+        	ui->passwordLineEdit->setEnabled(false);
+
+            if(ui->zoomSpinBox->value() > 20) {
```

Comment:
That is reasonable -- my original comment was partially about future proofing.  (Also, I experimented, the clamping is automatically performed when the maximum is changed.)

But I like what @avitebskiy said better.  However, this means that the zoom level cannot be set until the connection is tested.  You might consider adding a "connect to source" button which unlocks zoom..